proposed/local-sdk-global-json.md

```diff
@@ -0,0 +1,215 @@
+failed the resolver would display the contents of `errorMessage` instead of
+the default error message.
+
+## Motivation
+
+There is currently a disconnect between the ways the .NET SDK is deployed in
+practice and what the host resolver can discover when searching for compatible
+SDKs. By default the host resolver is only going to search for SDKs in machine
+wide locations: `C:\Program Files\dotnet`, `%PATH%`, etc ...  The .NET SDK
+though is commonly deployed to local locations: `%LocalAppData%\Microsoft\dotnet`,
+`$HOME/.dotnet`. Many repos embrace this and restore the correct .NET for their
+builds into a local `.dotnet` directory.
+
+The behavior of the host resolver is incompatible with local based deployments.
+It will not find these deployments without additional environment variable
+configuration and instead search in machine wide locations. That means tools
+like Visual Studio and VS Code simply do not work with local deployment by
+default. Developers must take additional steps like manipulating `%PATH%` before
+launching these editors. That reduces the usefulness of items like the quick
+launch bar, short cuts, etc.
+
+This is further complicated when developers mix local and machine wide
+installations. The host resolver will find the first `dotnet` according to its
+lookup rules and search only there for a compatible SDK. Once developers
+manipulate `%PATH%` to prefer local SDKS the resolver will stop considering
+machine wide SDKS. That can lead to situations where there is machine wide SDK
+that works for a given global.json but the host resolver will not consider it
+because the developer setup `%PATH%` to consider a locally installed SDK. That
+can be very frustrating for end users.
+
+This disconnect between the resolver and deployment has lead to customers
+introducing a number of creative work arounds:
+
+- [scripts][example-scripts-razor] to launch VS Code while considering locally
+deployed .NET SDKs
+- [docs and scripts][example-scripts-build] to setup the environment and launch
+VS so it can find the deployed .NET SDKs.
+- [scripts][example-scripts-dotnet] that wrap `dotnet` to find the  _correct_
+`dotnet` to use during build.
+
+These scripts are not one offs, they are increasingly common items in repos in
+`github.com/dotnet` to attempt to fix the disconnect. Even so many of these
+solutions are incomplete because they themselves only consider local deployment.
+They dont't fully support the full set of ways the SDK can be deployed.
+
+This problem also manifests in how customers naturally want to use our
+development tools like Visual Studio or VS Code. It's felt sharply on the .NET
+team, or any external customer who wants to contribute to .NET, due to how
+arcade infrastructure uses xcopy deployment into `.dotnet`. External teams
+like Unity also feel this pain in their development:
+
+- This [issue][cases-sdk-issue] from 2017 attempting
+to solve this problsem. It gets several hits a year from customers who are
```

Comment:
```suggestion
to solve this problem. It gets several hits a year from customers who are
```

Reply:
Hmm ... is this my problsem or your problsem?